emmet-core/emmet/core/openmm/tasks.py

```diff
@@ -0,0 +1,217 @@
+from emmet.core.vasp.task_valid import TaskState  # type: ignore[import-untyped]
+from pydantic import BaseModel, Field
+
+from atomate2.classical_md.schemas import ClassicalMDTaskDocument
+
+
+class CalculationInput(BaseModel, extra="allow"):  # type: ignore[call-arg]
+    """OpenMM input settings for a job, these are the attributes of the OpenMMMaker."""
+
+    n_steps: Optional[int] = Field(
+        None, description="The number of simulation steps to run."
+    )
+
+    step_size: Optional[float] = Field(
+        None, description="The size of each simulation step (picoseconds)."
+    )
+
+    temperature: Optional[float] = Field(
+        None, description="The simulation temperature (kelvin)."
+    )
+
+    friction_coefficient: Optional[float] = Field(
+        None,
+        description=(
+            "The friction coefficient for the integrator " "(inverse picoseconds)."
+        ),
+    )
+
+    platform_name: Optional[str] = Field(
+        None,
+        description=(
+            "The name of the OpenMM platform to use, passed to "
+            "Interchange.to_openmm_simulation."
+        ),
+    )
+
+    platform_properties: Optional[dict] = Field(
+        None,
+        description=(
+            "Properties for the OpenMM platform, passed to "
+            "Interchange.to_openmm_simulation."
+        ),
+    )
+
+    state_interval: Optional[int] = Field(
+        None,
+        description=(
+            "The interval for saving simulation state. For no state, set to 0."
+        ),
+    )
+
+    state_file_name: Optional[str] = Field(
+        None, description="The name of the state file to save."
+    )
+
+    traj_interval: Optional[int] = Field(
+        None,
+        description=(
+            "The interval for saving trajectory frames. For no trajectory, set to 0."
+        ),
+    )
+
+    wrap_traj: Optional[bool] = Field(
+        None, description="Whether to wrap trajectory coordinates."
+    )
+
+    report_velocities: Optional[bool] = Field(
+        None, description="Whether to report velocities in the trajectory file."
+    )
+
+    traj_file_name: Optional[str] = Field(
+        None, description="The name of the trajectory file to save."
+    )
+
+    traj_file_type: Optional[str] = Field(
+        None,
+        description="The type of trajectory file to save. Options are 'dcd' and 'h5'.",
+    )
+
+
+class CalculationOutput(BaseModel):
+    """OpenMM calculation output files and extracted data."""
+
+    dir_name: Optional[str] = Field(
+        None, description="The directory for this OpenMM task"
+    )
+
+    dcd_file: Optional[str] = Field(
+        None, description="Path to the DCD file relative to `dir_name`"
+    )
+
+    state_file: Optional[str] = Field(
+        None, description="Path to the state file relative to `dir_name`"
+    )
+
+    steps: Optional[list[int]] = Field(
+        None, description="Steps where outputs are reported"
+    )
```

Comment:
`n_steps` in input vs. `steps` in output might be slightly confusing. Would it make sense to use a name like `save_every_steps` or something to be a little more clear?